### PR TITLE
add hdzero h7 target

### PR DESCRIPTION
**Pull request by freasy**
Add HDZERO H7 target

### CODE REVIEW

File: configs/HDZEROH743R/config.h [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+#define TIMUP8_DMA_OPT 0
+
+#define SERIALRX_UART SERIAL_PORT_USART1
+#define DEFAULT_BLACKBOX_DEVICE BLACKBOX_DEVICE_FLASH
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE 100
+#define BEEPER_INVERTED
+#define FLASH_SPI_INSTANCE SPI2
+#define USE_SPI_GYRO
```
Comment (haslinghuis) [inline configs/HDZEROH743R/config.h:104]: This is default
```suggestion
```

File: configs/HDZEROH743R/config.h [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+#define GYRO_1_SPI_INSTANCE SPI1
+#define GYRO_2_SPI_INSTANCE SPI1
```
Comment (haslinghuis) [inline configs/HDZEROH743R/config.h:106]: No return on EOF
```suggestion
#define GYRO_2_SPI_INSTANCE SPI1

```

**Review by haslinghuis — CHANGES_REQUESTED** [review]

File: configs/HDZEROH743R/config.h [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+#define GYRO_1_EXTI_PIN PC13
+#define GYRO_1_CS_PIN PB4
+#define GYRO_2_EXTI_PIN PA15
+#define GYRO_2_CS_PIN PB3
+
+#define TIMER_PIN_MAPPING \
+    TIMER_PIN_MAP(  0, MOTOR1_PIN,  1,  0 ) \
+    TIMER_PIN_MAP(  1, MOTOR2_PIN,  1,  0 ) \
+    TIMER_PIN_MAP(  2, MOTOR3_PIN,  2,  0 ) \
+    TIMER_PIN_MAP(  3, MOTOR4_PIN,  2,  0 ) \
```
Comment (ot0tot) [inline configs/HDZEROH743R/config.h:81]: Why are these on TIM8? If you move motors 3 and 4 to TIM3, only one TIMUP resource is required for burst DMA.

File: configs/HDZEROH743R/config.h [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+#pragma once
+
+#define FC_TARGET_MCU     STM32H743
+
+#define BOARD_NAME        HDZEROH743R
+#define MANUFACTURER_ID   HDZO
+
+#define USE_GYRO
+#define USE_GYRO_SPI_MPU6000
+#define USE_GYRO_SPI_MPU6500
```
Comment (ot0tot) [inline configs/HDZEROH743R/config.h:31]: Why include MPU6500?

File: configs/HDZEROH743R/config.h [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+#define TIMER_PIN_MAPPING \
+    TIMER_PIN_MAP(  0, MOTOR1_PIN,  1,  0 ) \
+    TIMER_PIN_MAP(  1, MOTOR2_PIN,  1,  0 ) \
+    TIMER_PIN_MAP(  2, MOTOR3_PIN,  2,  0 ) \
+    TIMER_PIN_MAP(  3, MOTOR4_PIN,  2,  0 ) \
+    TIMER_PIN_MAP(  4, LED_STRIP_PIN,  1,  0 ) \
+    TIMER_PIN_MAP(  5, BEEPER_PIN,  1,  0 )
+
+#define SPI1_RX_DMA_OPT 7
```
Comment (ot0tot) [inline configs/HDZEROH743R/config.h:85]: Not needed

File: configs/HDZEROH743R/config.h [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+#define BOARD_NAME        HDZEROH743R
+#define MANUFACTURER_ID   HDZO
+
+#define USE_GYRO
+#define USE_GYRO_SPI_MPU6000
+#define USE_GYRO_SPI_MPU6500
+#define USE_GYRO_SPI_ICM42688P
+#define USE_ACC
+#define USE_ACC_SPI_MPU6000
+#define USE_ACC_SPI_MPU6500
```
Comment (ot0tot) [inline configs/HDZEROH743R/config.h:35]:
```suggestion

```

File: configs/HDZEROH743R/config.h (hunk 3 above)
Comment (ot0tot) [inline configs/HDZEROH743R/config.h:81]: Why are these on TIM8? If you move motors 3 and 4 to TIM3, only one TIMUP resource is required for burst DMA.
```suggestion
    TIMER_PIN_MAP(  2, MOTOR3_PIN,  1,  0 ) \
    TIMER_PIN_MAP(  3, MOTOR4_PIN,  1,  0 ) \
```

Comment (ot0tot) [inline configs/HDZEROH743R/config.h:81]: Why are these on TIM8? If you move motors 3 and 4 to TIM3, only one TIMUP resource is required for burst DMA.
```suggestion
    TIMER_PIN_MAP(  2, MOTOR3_PIN,  1,  0 ) \
    TIMER_PIN_MAP(  3, MOTOR4_PIN,  1,  0 ) \
```

File: configs/HDZEROH743R/config.h [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+#define TIMUP3_DMA_OPT 0
+#define TIMUP4_DMA_OPT 0
+#define TIMUP5_DMA_OPT 0
+#define TIMUP8_DMA_OPT 0
+
+#define SERIALRX_UART SERIAL_PORT_USART1
+#define DEFAULT_BLACKBOX_DEVICE BLACKBOX_DEVICE_FLASH
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE 100
```
Comment (ot0tot) [inline configs/HDZEROH743R/config.h:101]: Does this need to be set? Does this FC come with an ESC?

File: configs/HDZEROH743R/config.h (hunk 4 above)
Comment (sugaarK) [reply configs/HDZEROH743R/config.h:31]: I won't let them make a 6500 version!

**Comment by sugaarK** [comment]
@freasy can you address the changes? or are you waiting for samples?

**Comment by freasy** [comment]
> @freasy can you address the changes? or are you waiting for samples?

Samples are on the way

File: configs/HDZEROH743R/config.h [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+#define ADC2_DMA_OPT 9
+#define ADC3_DMA_OPT 10
+
+#define TIMUP1_DMA_OPT 0
+#define TIMUP3_DMA_OPT 0
+#define TIMUP4_DMA_OPT 0
+#define TIMUP5_DMA_OPT 0
+#define TIMUP8_DMA_OPT 0
+
+#define USE_TELEMETRY
```
Comment (haslinghuis) [inline configs/HDZEROH743R/config.h:91]: This is defined by RX protocol
```suggestion
```

**Review by haslinghuis — COMMENTED** [review]
No barometer ?

File: configs/HDZEROH743R/config.h [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+#define ADC1_DMA_OPT 8
+#define ADC2_DMA_OPT 9
+#define ADC3_DMA_OPT 10
+
+#define TIMUP1_DMA_OPT 0
+#define TIMUP3_DMA_OPT 0
+#define TIMUP4_DMA_OPT 0
+#define TIMUP5_DMA_OPT 0
+#define TIMUP8_DMA_OPT 0
```
Comment (ot0tot) [inline configs/HDZEROH743R/config.h:89]:
```suggestion
#define TIMER_PIN_MAPPING \
    TIMER_PIN_MAP(  0, MOTOR1_PIN,  1,  0 ) /* TIM3 CH1 */ \
    TIMER_PIN_MAP(  1, MOTOR2_PIN,  1,  1 ) /* TIM3 CH2 */ \
    TIMER_PIN_MAP(  2, MOTOR3_PIN,  1,  2 ) /* TIM3 CH3 */ \
    TIMER_PIN_MAP(  3, MOTOR4_PIN,  1,  3 ) /* TIM3 CH4 */ \
    TIMER_PIN_MAP(  4, LED_STRIP_PIN,  1,  4 ) /* TIM1 CH3 */ \
    TIMER_PIN_MAP(  5, BEEPER_PIN,  1,  -1) /* TIM4 CH1 */

#define ADC1_DMA_OPT 8
#define ADC2_DMA_OPT 9
#define ADC3_DMA_OPT 10

#define TIMUP1_DMA_OPT 0
#define TIMUP3_DMA_OPT 5
#define TIMUP4_DMA_OPT 0
#define TIMUP5_DMA_OPT 0
#define TIMUP8_DMA_OPT 0
```

File: configs/HDZEROH743R/config.h [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+#define UART2_RX_PIN PA3
+#define UART3_RX_PIN PB11
+#define UART4_RX_PIN PA1
+#define UART5_RX_PIN PD2
+#define UART7_RX_PIN PE7
+#define UART8_RX_PIN PE0
+#define I2C1_SCL_PIN PB8
+#define I2C1_SDA_PIN PB9
+#define LED0_PIN PE2
+#define LED1_PIN PC14
```
Comment (ot0tot) [inline configs/HDZEROH743R/config.h:57]: PC14 should not be used to drive LEDs per the datasheet

**Review by haslinghuis — APPROVED** [review]

File: configs/HDZEROH743R/config.h [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+#define GYRO_2_CS_PIN PB3
+
+#define TIMER_PIN_MAPPING \
+    TIMER_PIN_MAP(  0, MOTOR1_PIN,  1,  0 ) /* TIM3 CH1 */ \
+    TIMER_PIN_MAP(  1, MOTOR2_PIN,  1,  1 ) /* TIM3 CH2 */ \
+    TIMER_PIN_MAP(  2, MOTOR3_PIN,  1,  2 ) /* TIM3 CH3 */ \
+    TIMER_PIN_MAP(  3, MOTOR4_PIN,  1,  3 ) /* TIM3 CH4 */ \
+    TIMER_PIN_MAP(  4, LED_STRIP_PIN,  1,  4 ) /* TIM1 CH3 */ \
+    TIMER_PIN_MAP(  5, BEEPER_PIN,  1,  -1 ) /* TIM4 CH1 */
+
```
Comment (ot0tot) [inline configs/HDZEROH743R/config.h:79]: Include TIMUP DMA assignment so burst DMA works also. 
```suggestion

#define TIMUP3_DMA_OPT      5

```

**Comment by ot0tot** [comment]
Why are there two gyros defined? It appears the FC only has one gyro?

**Comment by sugaarK** [comment]
> Why are there two gyros defined? It appears the FC only has one gyro?

Comes in 2 flavours

**Pull request merged**